lib/phoenix/channel/driver.ex

```diff
@@ -0,0 +1,227 @@
+# TODO(sj): move the file to the proper place
+defmodule Phoenix.Socket.Driver do
+  @moduledoc """
+  Implementation of the Phoenix Channels protocol.
+
+  The Phoenix Channels protocol describes how to hold multiple separate conversations
+  (channels) over a single persistent connection (socket). The connection is
+  powered by transports, such as websocket or long polling, while this module
+  provides the stock implementation of the protocol, with following properties:
+
+  - Each channel runs in a separate process.
+  - Channel processes are direct children of the socket process.
+  - Termination of a channel process doesn't affect other channel processes or
+    the socket process.
+  - If the socket process terminates, all channel processes will be stopped as well,
+    regardless of the exit reason.
+
+  This module was not meant to be used directly. Instead you can pass it as an option
+  to the transport (TODO(sj): explain how, once we settle on the approach). This
+  happens by default when your module is powered by `Phoenix.Socket`.
+  """
+  @behaviour Phoenix.Transports.Driver
+
+  require Logger
+  alias Phoenix.Socket
+  alias Phoenix.Socket.Broadcast
+  alias Phoenix.Socket.Message
+  alias Phoenix.Socket.Reply
+
+  @client_vsn_requirements "~> 1.0"
+  @protocol_version "1.0.0"
+
+  @doc """
+  Returns the Channel Transport protocol version.
+  """
+  # TODO(sj): do we really need to expose this? It seems to be used only in tests.
```

Comment:
I think it is nice for tools or anything that want to show the protocol version from the elixir side.